modulefiles/build_hera_gnu.lua

```diff
@@ -19,6 +20,10 @@ load(pathJoin("nccmp", os.getenv("nccmp_ver") or "1.9.0.1"))
 load(pathJoin("nco", os.getenv("nco_ver") or "5.0.6"))
 load(pathJoin("openblas", os.getenv("openblas_ver") or "0.3.19"))
 
+prepend_path("CPPFLAGS", " -I/apps/slurm_hera/23.11.3/include/slurm"," ")
+prepend_path("LD_LIBRARY_PATH", "/apps/slurm_hera/23.11.3/lib")
+setenv("LD_PRELOAD", "/scratch2/NCEPDEV/stmp1/role.epic/installs/gnu/13.3.0/lib64/libstdc++.so.6")
```

Comment:
For the `CPPFLAGS` and `LD_LIBRARY_PATH` entries, is there a reason why these need to be added to the `build_hera_gnu.lua` modulefile?  I see the `LD_PRELOAD` pointing to a specific version of `libstdc++.so.6`.  Is this due to similar issues encountered on Gaea?

Since neither `Slurm` or `LD_PRELOAD` should need to be exercised during the build phase, should we add a `wflow_hera_gnu.lua` (and rename `wflow_hera.lua` to `wflow_hera_intel.lua`) to include these (I understand that we wouldn't want these lines in the `wflow_hera.lua file`, since these are specific to GNU).

Reply:
@MichaelLueken - yes, the LD_LIBRARY_PATH and LD_PRELOAD are needed during the run. However, the build_hera_gnu.lua is loaded during each task execution. So LD_LIBRARY_PATH=/apps/slurm_hera/23.11.3/lib had to be prepended and be in the front of other libs to find the correct slurm library, and not a default path.
 A preloaded libstdc++.so is needed to avoid libstdc++.so from the installed conda being used. When the wflow_hera is loaded and  conda env used, is finds the libstdc++.so in conda with a lower version of a GLIBCXX string than required, which is the version  used during SRW model build/compilation. An explicit preloading of the libstdc++.so from the gnu/13.3.0 is needed to overcome this conda library path.
 
 Issue with the libstc++.so might be similar to what we had on Gaea earlier. I'm not sure that the most recent issue with Gaea had the same problem (haven't got to it yet), but it may have a similar solution with placing LD_LIBRARY_PATH and/or LD_PRELOAD in the build_gaea_intel.gnu module!

Reply:
@natalie-perlin - Thanks for the information!  What you have done makes sense.  I'll allow other reviewers to weigh in, if they so desire, but I won't keep this from allowing this PR to move forward, once the weather model has been updated to use GNU/13.3 on Hera.